src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -292,6 +333,35 @@ public sealed interface MemoryLayout permits SequenceLayout, GroupLayout, Paddin
      */
     MemoryLayout withByteAlignment(long byteAlignment);
 
+    /**
+     * {@return {@code offset + (byteSize() * index)}}
+     *
+     * @param offset the base offset
+     * @param index the index to be scaled by the byte size of this layout
+     */
+    @ForceInline
+    default long scale(long offset, long index) {
```

Comment:
was thinking... do we want the output of this method to ever be negative? (since it's used as an offset to a segment). Also, should we check for overflows with multiplyExact?

Reply:
I initially consider negative inputs, I wasn't sure, but thought they might be useful in some cases, so I left them in.

multiplyExact/addExtact seems like a good idea to use.

Reply:
If the output of `scale` is always used as an offset parameter in some VH, or memory segment accessor, then I think there's a very strong gravitational pull for making it positive. I can't think of any methods in the MemorySegment API that would be happy with a negative offset, so it seems like a place where bugs might hide, or be reported in unexpected places.

Reply:
That's a good point